### PR TITLE
Eliminate cv2 dependency from Grad-CAM example

### PR DESCRIPTION
**Pull request by swghosh**
The extra dependency for cv2 library have been eliminated by use of `plt.imshow` with `alpha` set according to required intensity instead of `cv2.applyColorMap` without affecting any functionality in the Grad CAM example as present.

This PR is intended at reducing build times for future versions by eliminating an extra package where an automatic CI/CD pipeline would be used.

### CODE REVIEW

File: examples/vision/md/grad_cam.md [hunk 1]
```diff
@@ -171,30 +177,28 @@ Predicted: [('n02504458', 'African_elephant', 0.8871446)]
...
 
 # 0.4 here is a heatmap intensity factor
-superimposed_img = heatmap * 0.4 + img
-
-# Save the image to disk
-save_path = "elephant_cam.jpg"
-cv2.imwrite(save_path, superimposed_img)
-
-display(Image(save_path))
+plt.imshow(heatmap, cmap="jet", alpha=0.4)
```
Comment (fchollet) [inline examples/vision/md/grad_cam.md:197]: IMO the visuals from calling `display(Image(save_path))` are much better than using `plt`. Nevertheless, it should be possible to remove the cv2 dependency while still using `display(Image())`. We could do use `array_to_img` to create a PIL image holding the superimposed images, then use `img.save(path)` to save it to disk.

Comment (swghosh) [reply examples/vision/md/grad_cam.md:197]: I'll make the changes and get back to you soon. Thanks for reviewing this PR.

Comment (swghosh) [reply examples/vision/md/grad_cam.md:197]: Should I use something like a more minimal version of [this](https://gist.github.com/jimfleming/c1adfdb0f526465c99409cc143dea97b) to apply jet color map on the heatmap for RGB format basis `matplotlib.cm` and then use `img = img + 0.4 * heatmap`?

Comment (fchollet) [reply examples/vision/md/grad_cam.md:197]: I haven't tried it, and I image there might be a simpler solution, but you could do something like

```python
img = keras.preprocessing.image.load_img(img_path)
img = keras.preprocessing.image.img_to_array(img)
heatmap = np.expand_dims(heatmap, axis=-1)
heatmap = heatmap.resize((img.shape[1], img.shape[0]))
heatmap = keras.preprocessing.image.array_to_img(heatmap)
heatpmap = keras.preprocessing.image.img_to_array(heatpmap)
superimposed_img = heatmap * 0.4 + img
superimposed_img = keras.preprocessing.image.array_to_img(superimposed_img)
superimposed_img.save(...)
Display(Image(...))
```

Comment (swghosh) [reply examples/vision/md/grad_cam.md:197]: I was trying something like:

```python3
# We rescale heatmap to a range 0-255
heatmap = np.uint8(255 * heatmap)

# Colorize the heatmap to jet 
jet = cm.get_cmap('jet')
jet_colors = jet(np.arange(256))
jet_heatmap = jet_colors[heatmap]

# We create an image from the heatmap
jet_heatmap = keras.preprocessing.image.array_to_img(jet_heatmap)
jet_heatmap = jet_heatmap.resize((img.shape[1], img.shape[0]))
jet_heatmap = keras.preprocessing.array_to_img(jet_heatmap)

# Create an overlay of the heatmap on original image
superimposed_img = jet_heatmap * 0.4 + img

superimposed_img = keras.preprocessing.image.array_to_img(superimposed_img)
superimposed_img.save(...)
Display(Image(...))
```

Which gives me something like this:
GRAD CAM with JET
)

Comment (swghosh) [reply examples/vision/md/grad_cam.md:197]: > I haven't tried it, and I imagine there might be a simpler solution, but you could do something like
> 
> ```python
> img = keras.preprocessing.image.load_img(img_path)
> img = keras.preprocessing.image.img_to_array(img)
> heatmap = np.expand_dims(heatmap, axis=-1)
> heatmap = heatmap.resize((img.shape[1], img.shape[0]))
> heatmap = keras.preprocessing.image.array_to_img(heatmap)
> heatpmap = keras.preprocessing.image.img_to_array(heatpmap)
> superimposed_img = heatmap * 0.4 + img
> superimposed_img = keras.preprocessing.image.array_to_img(superimposed_img)
> superimposed_img.save(...)
> Display(Image(...))
> ```

This is giving me something like this:
GRAD CAM without Jet cmap
As the grayscale heatmap is directly being used superimposed without application of the Jet colormap.

**Comment by swghosh** [comment]
I have updated the source code to use IPython display instead of plt (plus, applying the jet cmap using numpy and keras only eliminating cv2) so as to facilitate better visuals. Request to your review the changes on this PR.

**Pull request merged**